rest/route/stats.go

```diff
@@ -344,3 +374,97 @@ func (tsh *testStatsHandler) makeStartAtKey(testStats model.APITestStats) string
 func makeGetProjectTestStats(sc data.Connector) gimlet.RouteHandler {
 	return &testStatsHandler{sc: sc}
 }
+
+///////////////////////////////////////////////
+// /projects/<project_id>/task_stats handler //
+///////////////////////////////////////////////
+
+type taskStatsHandler struct {
+	sc data.Connector
+	statsHandler
+}
+
+func (tsh *taskStatsHandler) Factory() gimlet.RouteHandler {
+	return &taskStatsHandler{sc: tsh.sc}
+}
+
+func (tsh *taskStatsHandler) Parse(ctx context.Context, r *http.Request) error {
+	vals := r.URL.Query()
+	project := gimlet.GetVars(r)["project_id"]
+	tsh.filter = stats.StatsFilter{Project: project}
+
+	err := tsh.statsHandler.parseStatsFilter(vals)
+	if err != nil {
+		return err
+	}
+	err = tsh.filter.ValidateForTasks()
+	if err != nil {
+		return gimlet.ErrorResponse{
+			Message:    err.Error(),
+			StatusCode: http.StatusBadRequest,
+		}
+	}
+	return nil
+}
+
+func (tsh *taskStatsHandler) Run(ctx context.Context) gimlet.Responder {
+	var err error
+	var taskStatsResult []model.APITaskStats
+
+	taskStatsResult, err = tsh.sc.GetTaskStats(&tsh.filter)
+	if err != nil {
+		return gimlet.MakeJSONInternalErrorResponder(errors.Wrap(err, "Failed to retrieve the task stats"))
+	}
+
+	resp := gimlet.NewResponseBuilder()
+	if err = resp.SetFormat(gimlet.JSON); err != nil {
+		return gimlet.MakeJSONInternalErrorResponder(err)
+	}
+
+	requestLimit := tsh.filter.Limit - 1
+	lastIndex := len(taskStatsResult)
+	if len(taskStatsResult) > requestLimit {
+		lastIndex = requestLimit
+
+		err = resp.SetPages(&gimlet.ResponsePages{
+			Next: &gimlet.Page{
+				Relation:        "next",
+				LimitQueryParam: "limit",
+				KeyQueryParam:   "start_at",
+				BaseURL:         tsh.sc.GetURL(),
+				Key:             tsh.makeStartAtKey(taskStatsResult[requestLimit]),
+				Limit:           requestLimit,
+			},
+		})
+		if err != nil {
+			return gimlet.MakeJSONInternalErrorResponder(errors.Wrap(err,
+				"Problem paginating response"))
+		}
+	}
+	taskStatsResult = taskStatsResult[:lastIndex]
+
+	for _, apiTaskStats := range taskStatsResult {
+		if err = resp.AddData(apiTaskStats); err != nil {
+			return gimlet.MakeJSONInternalErrorResponder(err)
+		}
+	}
+
+	return resp
+}
+
+// makeStartAtKey creates a key string that can be used as a start_at value to fetch
+// the next results with pagination.
+func (tsh *taskStatsHandler) makeStartAtKey(taskStats model.APITaskStats) string {
+	elements := []string{
+		taskStats.Date,
+		taskStats.BuildVariant,
+		taskStats.TaskName,
+		"",
```

Comment:
why skip this key? there should be some documentation here? 

Perhaps it makes sense to have this be a method on model.APITaskStats?

Reply:
Good point. I added a comment.

Reply:
Makes sense. I added a struct in the model and we can now call StartAtKey() on APITestStats and APITaskStats.